### ISSUE TITLE
App crash in onCreateView ,  activity is null

### ISSUE THREAD

**Issue by TikTak123** [bluelinelabs/Conductor#595]
**Conductor version:**
implementation 'com.bluelinelabs:conductor:3.0.0-rc2'

**API level:**
Android 5, 6, 7, 8, 9

**How to reproduce it:**
I don’t know how to reproduce.  I have 40,000 users and 1% of them have this error. This statistic from Crashlytics.

**Error log:**
```
Fatal Exception: java.lang.NullPointerException: Cannot return null from a non-@Nullable @Provides method
       at dagger.internal.Preconditions.checkNotNull(Preconditions.java:48)
       at com.example.di.activity.ControllerModule_ProvideActivityContextFactory.provideActivityContext(ControllerModule_ProvideActivityContextFactory.java:29)
       at com.example.di.activity.ControllerModule_ProvideActivityContextFactory.get(ControllerModule_ProvideActivityContextFactory.java:21)
       at com.example.di.activity.ControllerModule_ProvideActivityContextFactory.get(ControllerModule_ProvideActivityContextFactory.java:8)
       at dagger.internal.DoubleCheck.get(DoubleCheck.java:47)
       at com.example.ui.main.main.TabMainViewModel_Factory.get(TabMainViewModel_Factory.java:95)
       at com.example.ui.main.main.TabMainViewModel_Factory.get(TabMainViewModel_Factory.java:23)
       at dagger.internal.DoubleCheck.get(DoubleCheck.java:47)
       at com.example.di.DaggerApplicationComponent$ControllerComponentImpl.injectTabMainController(DaggerApplicationComponent.java:1288)
       at com.example.di.DaggerApplicationComponent$ControllerComponentImpl.inject(DaggerApplicationComponent.java:977)
       at com.example.ui.main.main.TabMainController.onCreateView(TabMainController.java:76)
       at com.bluelinelabs.conductor.Controller.inflate(Controller.java:1013)
       at com.bluelinelabs.conductor.ControllerChangeHandler.executeChange(ControllerChangeHandler.java:190)
       at com.bluelinelabs.conductor.ControllerChangeHandler.executeChange(ControllerChangeHandler.java:154)
       at com.bluelinelabs.conductor.Router.performPendingControllerChanges(Router.java:831)
       at com.bluelinelabs.conductor.Router$3.run(Router.java:819)
       at android.os.Handler.handleCallback(Handler.java:873)
       at android.os.Handler.dispatchMessage(Handler.java:99)
       at android.os.Looper.loop(Looper.java:201)
       at android.app.ActivityThread.main(ActivityThread.java:6864)
       at java.lang.reflect.Method.invoke(Method.java)
       at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:547)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:873)

```

**Sample code:**
TabMainController.java
```
@NonNull
    @Override
    protected View onCreateView(@NonNull LayoutInflater inflater, @NonNull ViewGroup container) {
        BaseApplication.getComponent().controllerComponent(new ControllerModule(getActivity())).inject(this);
        binding = ControllerTabMainBinding.inflate(inflater, container, false);
        binding.setViewmodel(mViewModel);
        mViewModel.setCallback(this, getArgs());

        binding.swipeLayout.setColorSchemeResources(R.color.colorPrimary, R.color.black);
        binding.showHideDetails.setOnClickListener(null);

        return binding.getRoot();
    }
```

```
@Module
public class ControllerModule {

    protected final Activity activity;

    public ControllerModule(Activity activity) {
        this.activity = activity;
    }

    @Provides
    @PerController
    public Activity provideActivityContext() {
        return this.activity;
    }
}

```

**Comment by TikTak123** [comment]
I think this happens in activity restoration, but I'm not sure

**Comment by EricKuck** [comment]
I'm struggling to reproduce this one as well. To help narrow it down a bit, is `TabMainController` in an Activity hosted Router, or is it in a child Router of another Controller?

**Comment by TikTak123** [comment]
TabMainController in child Router of MainMenuController. Activity hosted Router is MainMenuController. Source code of MainMenuController you can see here https://github.com/bluelinelabs/Conductor/issues/596

**Comment by TikTak123** [comment]
After migration from implementation 'com.bluelinelabs:conductor:3.0.0-rc2'
to implementation 'com.bluelinelabs:conductor:3.0.1-SNAPSHOT' this crash doesn't reproduce
I don't see crash logs in crashlytics about this.

**Comment by TikTak123** [comment]
Thank you

**Comment by HauLuu** [comment]
@TikTak123 
I'm also facing this problem. The activity is null some time in onCreateView. 

by implementation 'com.bluelinelabs:conductor:3.0.1-SNAPSHOT'  do you mean implementation 'com.bluelinelabs:conductor:3.0.0-rc5'?

**Comment by TikTak123** [comment]
I don't remember in which version is fixed. I think it fixed in 3.0.0-rc3 version. Now I'm using 3.0.0-rc6, it is OK about this.

**Comment by HauLuu** [comment]
I'm using 3.0.0-rc5 now. Not sure this issue is fixed or not, I think I just need to wait for a while to see is there any similar report coming. 

I noticed one thing after upgrading from 3.0.0-rc1 to 3.0.0-rc5, the onCreateView method signature has changed

from 
fun onCreateView(inflater: LayoutInflater, container: ViewGroup): View

to 
fun onCreateView(inflater: LayoutInflater, container: ViewGroup, savedViewState: Bundle?): View
